pkg/storage/mongo/lister.go

```diff
@@ -4,28 +4,37 @@ import (
 	"context"
 
 	"github.com/globalsign/mgo"
-	"github.com/globalsign/mgo/bson"
 	"github.com/gomods/athens/pkg/errors"
 	"github.com/gomods/athens/pkg/observ"
 	"github.com/gomods/athens/pkg/storage"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
 // List lists all versions of a module
 func (s *ModuleStore) List(ctx context.Context, module string) ([]string, error) {
 	const op errors.Op = "mongo.List"
 	ctx, span := observ.StartSpan(ctx, op.String())
 	defer span.End()
-	c := s.s.DB(s.d).C(s.c)
-	fields := bson.M{"version": 1}
-	result := make([]storage.Module, 0)
-	err := c.Find(bson.M{"module": module}).Select(fields).All(&result)
+	c := s.client.Database(s.db).Collection(s.coll)
+	projection := bson.M{"version": 1}
+	query := bson.E{Key: "module", Value: module}
+	cursor, err := c.Find(context.Background(), query, &options.FindOptions{Projection: projection})
 	if err != nil {
 		kind := errors.KindUnexpected
 		if err == mgo.ErrNotFound {
 			kind = errors.KindNotFound
 		}
 		return nil, errors.E(op, kind, errors.M(module), err)
 	}
+	result := make([]storage.Module, 0)
+	for cursor.Next(context.Background()) {
+		var module storage.Module
+		err := cursor.Decode(&module)
+		if err == nil {
```

Comment:
As above - we should not ignore errors. If we do List could provide different results at t1 and t2.